worker/broker.go

```diff
@@ -0,0 +1,510 @@
+/*
+ * Copyright 2019 The CovenantSQL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package worker
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"strings"
+	"time"
+
+	"path"
+
+	"sync"
+
+	"github.com/CovenantSQL/CovenantSQL/conf"
+	"github.com/CovenantSQL/CovenantSQL/crypto/kms"
+	"github.com/CovenantSQL/CovenantSQL/proto"
+	"github.com/CovenantSQL/CovenantSQL/types"
+	"github.com/CovenantSQL/CovenantSQL/utils/log"
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	"github.com/gogf/gf/g/container/gqueue"
+)
+
+type MQTTAPI string
+
+const (
+	//Publish API
+	MQTTNewest MQTTAPI = "newest"
+	DSNList    MQTTAPI = "dsnlist"
+
+	//Subscribe API
+	MQTTWrite  MQTTAPI = "write"
+	MQTTReplay MQTTAPI = "replay"
+	MQTTCreate MQTTAPI = "create"
+
+	MQTTInvalid MQTTAPI = ""
+)
+
+var (
+	publishPrefix = "/cql/miner/"
+	listenPrefix  = "/cql/client/"
+)
+
+type MQTTTypeOfValue int
+
+const (
+	MQTTNull MQTTTypeOfValue = iota
+	MQTTString
+	MQTTInt64
+	MQTTBool
+	MQTTFloat64
+	MQTTByte
+	MQTTTime
+)
+
+// NamedArgWithType defines the named argument structure for database, and add a type column for json decode.
+type NamedArgWithType struct {
+	Name  string
+	Value interface{}
+	Type  MQTTTypeOfValue
+}
+
+// Query defines single query.
+type MQTTQuery struct {
+	Pattern string
+	Args    []NamedArgWithType
+}
+
+type BrokerPayload struct {
+	BlockID        int32        `json:"block_id"`
+	BlockIndex     int          `json:"block_index"`
+	ClientID       proto.NodeID `json:"client_id"`
+	ClientSequence uint64       `json:"client_seq"`
+	Events         []MQTTQuery  `json:"events"`
+
+	//Replay API
+	BlockStart int32 `json:"block_start"`
+	IndexStart int   `json:"index_start"`
+	BlockEnd   int32 `json:"block_end"`
+	IndexEnd   int   `json:"index_end"`
+}
+
+type SubscribeEvent struct {
+	ClientID   proto.NodeID
+	DatabaseID proto.DatabaseID
+	ApiName    MQTTAPI
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
struct field ApiName should be APIName